src/hdr/encoder.rs

```diff
@@ -32,15 +32,15 @@ impl<W: Write> HDREncoder<W> {
             // new RLE marker contains scanline width
             let marker = rgbe8(2, 2, (width / 256) as u8, (width % 256) as u8);
             // buffers for encoded pixels
-            let mut bufr = Vec::with_capacity(width);
+            let mut bufr = vec![0; width];
             bufr.resize(width, 0);
```

Comment:
Since the buffer is directly allocated, `resize` should be removed. (3 more instances below).